### PR TITLE
feat(pie-modal): DSW-1430 add pie-spinner to pie-modal

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)
[Changed] - reuse pie-spinner in pie-modal

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-modal/package.json [hunk 1]
```diff
@@ -43,6 +38,12 @@
     "extends": "../../../package.json"
   },
   "dependencies": {
+    "@justeat/pie-design-tokens": "5.8.2",
+    "@justeattakeaway/pie-button": "0.36.0",
+    "@justeattakeaway/pie-icon-button": "0.21.0",
+    "@justeattakeaway/pie-icons-webc": "0.11.1",
+    "@justeattakeaway/pie-spinner": "0.1.0",
+    "@justeattakeaway/pie-webc-core": "0.11.0",
```
Comment (fernandofranca) [inline packages/components/pie-modal/package.json:46]: Why are we moving these? 🤔

**Review by fernandofranca — APPROVED** [review]

File: packages/components/pie-modal/package.json (hunk 1 above)
Comment (raoufswe) [reply packages/components/pie-modal/package.json:46]: I don't believe having them as a `devDependency` is the correct thing here. these dependencies are needed for the component to function and have to be shipped in the production bundle and, are not only needed for development purposes such as testing or storybook-related stuff.

**Review by kevinrodrigues — APPROVED** [review]

File: packages/components/pie-modal/src/index.ts [hunk 2]
```diff
@@ -342,6 +343,7 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
             <div class="c-modal-contentInner">
                 <slot></slot>
             </div>
+            ${this.isLoading ? html`<pie-spinner size="xl" variant="secondary"></pie-spinner>` : nothing}
```
Comment (kevinrodrigues) [inline packages/components/pie-modal/src/index.ts:346]: Tests?

Comment (raoufswe) [reply packages/components/pie-modal/src/index.ts:346]: we already have both visual and component tests in place, this PR just refactors the implementation of the behaviour.

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**